### ISSUE TITLE
Rule proposal: warn about missing dollar sign surrounding curly braces in template string

### ISSUE THREAD

**Issue by cmdcolin** [sindresorhus/eslint-plugin-unicorn#2150]
### Description

I migrated some code from React to template strings, but missed adding the $ sign around the curly braces so I had

`Hello {name}` instead of `Hello ${name}`

I did not see any existing eslint rules that warn against this. I also wasn't sure what the best rule set to propose this rule to, but I like eslint-plugin-unicorn a lot and wasn't sure this quite had the 'universality' that it would be a core eslint rule.

Somewhat similar rule in eslint core rule set https://eslint.org/docs/latest/rules/no-template-curly-in-string#examples

They say  when proposing a new rule to eslint core to justify e.g. "Why should this rule be in the core instead of a plugin?

In general, we prefer that rules be implemented in plugins where they can be tailored to your specific use case.". not sure this rule has enough justification?

### Fail

```js
var replace = `Hello {name}`;
```

### Pass

```js
const replace = `Hello ${name}`;
```

**Comment by sindresorhus** [comment]
Can you also propose a rule name?

**Comment by cmdcolin** [comment]
unicorn/missing-template-dollar-sign maybe?

**Comment by sindresorhus** [comment]
How about `unicorn/no-incorrect-template-string-interpolation`?

**Comment by fisker** [comment]
`no-mismatched-template-literal-curly`?

**Comment by fisker** [comment]
We can also check

```js
var replace = `Hello $name}`;
var replace = `Hello {name`;
```

**Comment by sindresorhus** [comment]
> no-mismatched-template-literal-curly?

I think `no-incorrect-template-string-interpolation` clearer as the intent is to catch mistakes in template string interpolation.

**Comment by cmdcolin** [comment]
thank you! I just ran into another one of these today :)
